libraries/AP_Networking/AP_Networking_PPP.cpp

```diff
@@ -99,14 +101,17 @@ void AP_Networking_PPP::ppp_status_callback(struct ppp_pcb_s *pcb, int code, voi
     }
     case PPPERR_PEERDEAD: {        /* Connection timeout */
         msg = "PPPERR_PEERDEAD";
+        driver.need_restart = true;
         break;
     }
     case PPPERR_IDLETIMEOUT: {     /* Idle Timeout */
         msg = "PPPERR_IDLETIMEOUT";
+        driver.need_restart = true;
         break;
     }
     case PPPERR_CONNECTTIME: {     /* Max connect time reached */
         msg = "PPPERR_CONNECTTIME";
+        driver.need_restart = true;
```

Comment:
There should be no message for this one I think.

Reply:
i've changed the unexpected ones to "PPP: code NN"

Reply:
That's much cleaner. Thanks Tridge.